library/MedEx/API.php

```diff
@@ -0,0 +1,2092 @@
+                                                    if ($prefs) {
+                                                        $facs = explode('|',$prefs['ME_facilities']);
+                                                        foreach ($facs as $place) {
+                                                            if ($place == $fac['id']) $checked = 'checked ="checked"';
+                                                        }
+                                                    }
+                                                    ?>
+                                                    <input <?php echo $checked; ?> class="update" type="checkbox" name="facilities[]" id="facility_<?php echo $fac['id']; ?>" value="<?php echo $fac['id']; ?>">
+                                                    <label for="facility_<?php echo $fac['id']; ?>"><?php echo $fac['name']; ?></label><br /><?php
+                                                }
+                                                ?>
+                                                </div>
+                                            </div>
+                                            <div class="divTableRow">
+                                                <div class="divTableCell divTableHeading"><?php echo xlt('Process Reminders for Provider'); ?></div>
+                                                <div class="divTableCell indent20">
+                                                <?php
+                                                $count="1";
+                                                $ures = sqlStatement("SELECT * FROM users WHERE authorized != 0 AND active = 1 ORDER BY lname, fname");
+                                                while ($prov = sqlFetchArray($ures))   {
+                                                    $checked ="";
+                                                    $suffix="";
+                                                    if ($prefs) {
+                                                        $provs = explode('|',$prefs['ME_providers']);
+                                                        foreach ($provs as $doc) {
+                                                            if ($doc == $prov['id']) $checked = 'checked ="checked"';
+                                                        }
+                                                    }
+                                                    if (!empty($prov['suffix']))  $suffix = ', '.$prov['suffix'];
+                                                    ?>
+                                                    <input <?php echo $checked; ?> class="update" type="checkbox" name="providers[]" id="provider_<?php echo $prov['id']; ?>" value="<?php echo $prov['id']; ?>">
+                                                    <label for="provider_<?php echo $prov['id']; ?>"><?php echo $prov['fname']." ".$prov['lname'].$suffix; ?></label><br /><?php
```

Comment:
ensure html escape. for example, above 2 lines are missing html escaping (via attr() and text() functions)